File: binary_chop_dailies/chop5.py
def binary_search(list, target):
    if len(list) == 0:
        return False
    else:
        mid_point = (len(list))//2
        if list[mid_point] == target:
            return True
        else:
            if list[mid_point] < target:
                return binary_search(list[mid_point+1:], target)
            else:
                return binary_search(list[:mid_point], target)

def rbs(ls, t):
    if not ls:
        return False
    else:
        mid_point = (len(ls))//2
        if ls[mid_point] == t:
            return True
        elif ls[mid_point] < t:
            return rbs(ls[mid_point+1:], t)
        else:
            return rbs(ls[:mid_point],t)

File: WED/binary_chop_dailies/chop5.rb
class Binary_Search
  define_method(:recursive_binary_search) do |list, target|
    return false if list.empty?

    mid = list.length / 2
    if list[mid] == target
      true
    elsif list[mid] >= target
      recursive_binary_search(list[0...mid], target)
    else
      recursive_binary_search(list[(mid + 1)..-1], target)
    end
  end
end

searcher = Binary_Search.new
numbers = (1..999).to_a
puts searcher.recursive_binary_search(numbers, 66)
puts searcher.recursive_binary_search(numbers, 9909)
